ConfigSpace/hyperparameters.py

```diff
@@ -30,36 +30,123 @@
 import warnings
 
 from collections import OrderedDict
-
-import numpy as np
+from typing import List, Any, Dict, Union, Tuple
 import io
-from functools import reduce
+import numpy as np
+
+
```

Comment:
Actually, all these methods are specific to one class, right? If yes, could you please move them to the respective classes?